File: Parcial1/punto4/punto4.c
#include <stdio.h>
#include <stdlib.h>
#include <time.h>

void swap(int *xp, int *yp)
{
    int temp = *xp;
    *xp = *yp;
    *yp = temp;
}

void bubbleSort(int arr[], int n)
{
    if (n == 1)
        return;

    int count = 0;
    for (int i=0; i<n-1; i++)
        if (arr[i] > arr[i+1]){
            swap(&arr[i], &arr[i+1]);
            count++;
        }

        if (count==0)
           return;

    bubbleSort(arr, n-1);
}

int main() {
    int n = 20000;
    int *arr = malloc(n * sizeof(int));
    srand(time(NULL));

    for (int i = 0; i < n; i++) {
        arr[i] = rand() % 100000;
    }

    clock_t inicio = clock();
    bubbleSort(arr, n);
    clock_t fin = clock();

    double tiempo = (double)(fin - inicio) / CLOCKS_PER_SEC;
    printf("Ordenamiento en C con %d elementos: %f segundos\n", n, tiempo);

    free(arr);
    return 0;
}

File: Parcial1/punto4/punto4.py
import random
import time

class bubbleSort:

    def __init__(self, array):
        self.array = array
        self.length = len(array)

    def __str__(self):
        return " ".join([str(x) 
                        for x in self.array])

    def bubbleSortRecursive(self, n=None):
        if n is None:
            n = self.length
        count = 0

        if n == 1:
            return
        for i in range(n - 1):
            if self.array[i] > self.array[i + 1]:
                self.array[i], self.array[i +
                1] = self.array[i + 1], self.array[i]
                count = count + 1
        if (count==0):
            return

        self.bubbleSortRecursive(n - 1)

def main():
    n = 20000
    arr = [random.randint(0, 100000) for _ in range(n)]
    
    sort = bubbleSort(array)
    
    sort.bubbleSortRecursive()
    print("Sorted array :\n", sort)

    inicio = time.time()
    bubbleSort(arr)
    fin = time.time()

    print(f"Ordenamiento en Python con {n} elementos: {fin - inicio} segundos")


if __name__ == "__main__":
    main()





